Source/UI/Controllers/AccountController.cs

```diff
@@ -22,589 +22,606 @@
-		{
-			return View();
-		}
-	}
+            }
+
+            // If we got this far, something failed, redisplay form
+            return View(MVC.Account.Views.Register, model);
+        }
+
+        [UserContext(RequiresGamingGroup = false)]
+        [AllowAnonymous]
+        public virtual ActionResult ConsumeInvitation(string id, ApplicationUser currentUser)
+        {
+            AddUserToGamingGroupResult result = gamingGroupInvitationConsumer.AddExistingUserToGamingGroup(id);
+
+            if (result.UserAddedToExistingGamingGroup)
+            {
+                return RedirectToAction(MVC.GamingGroup.ActionNames.Index, MVC.GamingGroup.Name);
+            }
+
+            RegisterViewModel registerViewModel = new RegisterViewModel
+            {
+                EmailAddress = result.EmailAddress,
+                GamingGroupInvitationId = id
+            };
+
+            return this.View(MVC.Account.Views.RegisterAgainstExistingGamingGroup, registerViewModel);
+        }
+
+        //
+        // POST: /Account/Disassociate
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public virtual async Task<ActionResult> Disassociate(string loginProvider, string providerKey)
+        {
+            ManageMessageId? message = null;
+            IdentityResult result = await userManager.RemoveLoginAsync(User.Identity.GetUserId(), new UserLoginInfo(loginProvider, providerKey));
+            if (result.Succeeded)
+            {
+                message = ManageMessageId.RemoveLoginSuccess;
+            }
+            else
+            {
+                message = ManageMessageId.Error;
+            }
+            return RedirectToAction("Manage", new { Message = message });
+        }
+
+        //
+        // GET: /Account/Manage
+        public virtual ActionResult Manage(ManageMessageId? message)
+        {
+            ViewBag.StatusMessage =
+                message == ManageMessageId.SetPasswordSuccess ? "Your password has been set."
+                : message == ManageMessageId.ChangePasswordSuccess ? "Your password has been changed."
+                : message == ManageMessageId.ChangeEmailSuccess ? "Your email has been changed."
+                : message == ManageMessageId.RemoveLoginSuccess ? "The external login was removed."
+                : message == ManageMessageId.Error ? "An error occurred."
+                : "";
+
+            SetViewBag();
+            ManageAccountViewModel viewModel = GetBaseManageAccountViewModel();
+
+            return View(MVC.Account.Views.Manage, viewModel);
+        }
+
+        public virtual ActionResult CreateGamingGroup()
+        {
+            return View(MVC.Account.Views.CreateGamingGroup);
+        }
+
+        [Authorize]
+        [UserContext]
+        public virtual ActionResult UserGamingGroups(ApplicationUser currentUser)
+        {
+            var gamingGroups = _gamingGroupRetriever.GetGamingGroupsForUser(currentUser);

```

Comment:
Would it make more sense just to get the current gaming group instead of returning the list? We could just put a simple wrapper method for GetGamingGroup(int gamingGroupId) over the dataContext.FindById(gamingGroupId). Just an idea.